agent/engine/docker_task_engine.go

```diff
@@ -1285,6 +1286,7 @@ func getFirelensLogConfig(task *apitask.Task, container *apicontainer.Container,
 	logConfig.Config[logDriverFluentdAddress] = fluentd
 	logConfig.Config[logDriverAsyncConnect] = strconv.FormatBool(true)
 	logConfig.Config[logDriverSubSecondPrecision] = strconv.FormatBool(true)
+	logConfig.Config[logDriverBufferLimit] = apitask.LogDriverBufferLimit
```

Comment:
I'm confused, this suggested its a field on the task structure, but then in the other file I don't see it added as a field in that struct...

Reply:
I learnt from this PR that just adding one field here is enough: https://github.com/aws/amazon-ecs-agent/pull/2538/files#diff-0f6cc891620e98230e92868ea887af5e1f11d9e692fc8415a50c1dd0b130cc8cR1106

Reply:
It is not a field on the task structure, it is a field in Config. And we need to get the value from the task package

Reply:
Okay, but our requirement is that each container can have its own log driver buffer setting. So I think it needs to be stored somewhere which is both per-task and per-container. Not some global setting.

Reply:
i don't think this needs to be saved anywhere. instead just do the following should work: (1) in collectFirelensLogOptions, skip adding the option to containerToLogOptions if the key equals firelensLogDriverBufferLimitOption; (2) here, get the actual value for "fluentd-buffer-limit" from the container's log config before we reset it at the previous line `logConfig.Config = make(map[string]string)`, and then save it to the config here.